### PR TITLE
Add Nats channel CRD implementation

### PR DESCRIPTION
**Pull request by montaro**
Fixes https://github.com/knative/eventing/issues/1567

**Release Note**
```release-note
Added Nats cluster channel provisioner implementation
```

### CODE REVIEW

File: natss/config/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+They do not offer:
+
+- Ordering guarantees
+  - Events seen downstream may not occur in the same order they were inserted
+    into the Channel.
+
+## Deployment steps
+
+1. Setup [Knative Eventing](../../../DEVELOPMENT.md).
```
Comment (vaikas-google) [inline natss/config/README.md]: Are these links still valid?

File: natss/config/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+1. If not done already, install a [NATS Streaming](./broker)
+1. Apply the NATSS configuration:
+
+   ```shell
+   ko apply -f contrib/natss/config
```
Comment (vaikas-google) [inline natss/config/README.md]: And this seems out of date?

File: natss/config/broker/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# NATS Streaming - simple installation
+
+1. For an installation of a simple NATS Streaming server, a setup is provided:
+
+   ```sbtshell
+   kubectl create namespace natss
+   kubectl apply -n natss -f contrib/natss/config/broker/natss.yaml
```
Comment (vaikas-google) [inline natss/config/broker/README.md]: I think this too needs updating?

File: natss/config/broker/natss.yaml [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# Copyright 2018 The Knative Authors
```
Comment (vaikas-google) [inline natss/config/broker/natss.yaml]: nit, 2019 (here and elsewhere)

**Comment by vaikas-google** [comment]
@n3wscott / @Harwayne / @nachocano what should the story be here for e2e tests?

**Comment by nachocano** [comment]
> @n3wscott / @Harwayne / @nachocano what should the story be here for e2e tests?

As you mentioned above, I think we should only move the CRD based channels, together with their e2e tests here. And leave the provisioner code and their e2e tests in eventing. We'll remove them next week after the 0.8 cut. 
We may need help from @Fredy-Z  for e2e test migration.

File: natss/config/provisioner/README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+They do not offer:
+
+- Ordering guarantees
+  - Events seen downstream may not occur in the same order they were inserted
+    into the Channel.
+
+## Deployment steps
+
+1. Setup [Knative Eventing](../../../../DEVELOPMENT.md).
+1. If not done already, install a [NATS Streaming](../broker)
```
[the remark this answers is not in the document]
Comment (montaro) [reply natss/config/provisioner/README.md]: I believe so

File: natss/config/README.md (hunk 1 above)
Comment (montaro) [reply natss/config/README.md]: will fix

**Comment by montaro** [comment]
@vaikas-google Hey, I fixed all the review comments, can you please review now?

**Comment by grantr** [comment]
Prow is blocking merge here because the OWNERS_ALIASES file doesn't contain the aliases referenced in this OWNERS file. #511 fixed that, and  supposedly `/verify-owners` should work to check that again. Most likely it would require merging master into this branch.

**Pull request merged**